config/core/configmaps/features.yaml

```diff
@@ -40,6 +40,13 @@ data:
     # this example block and unindented to be in the data block
     # to actually change the configuration.
 
+    # Default SecurityContext settings to secure-by-default values
+    # if unset.
```

Comment:
I'm curious what the compatibility guarantees of this are.  Personally, I'd love for this to evolve and _**change**_ as the best practices change, but that means minor version bumps could be breaking!  It would be good to link to some guidance we expect this flag to follow, so it seems less ad hoc.  Even a knative docs page would be better than nothing.

I suspect that stuff like disallowing privilege escalation and running as non-root as non-contentious parts of this, but I'm curious about potentially more contentious things like read-only root filesystems?

Reply:
Some projects consider seccompProfile possible risk https://github.com/istio/istio/issues/35894.

Reply:
I think we'll want to evolve and change this as well, assuming that Kubernetes continues to add "turn this knob to be secure" settings. One interesting challenge will be supporting new security settings across a range of Kubernetes API versions -- we may need to be more sophisticated about recognizing what fields need to be set for security.

I imagine if we end up needing to add more fields, we'll want to add new feature flags for those fields, e.g. `pod-security-defaults-k8s-1-27`, etc.

I'm also hoping / wishing that Kubernetes would pivot to a "secure by default" posture, but I understand that they have a lot of historical debt to pay down there (see `seccempProfile`, for example).